src/ILP/auth/ilp-credentials.ts

```diff
@@ -0,0 +1,25 @@
+import { Metadata } from 'grpc'
+import { AuthUtils } from './auth-utils'
+
+/**
+ * An extension of grpc.Metadata which provides a convenient way to
+ * add an Authorization metadata header, and ensures every bearer token
+ * going over the wire is prefixed with 'Bearer '
+ */
+class IlpCredentials extends Metadata {
+  /**
+   * Static initializer, which constructs a new IlpCredentials object and adds
+   * an Authorization entry.  If token is undefined, an Authorization header will
+   * still be added, but the call will ultimately fail as Unauthorized
+   *
+   * @param token an optional bearer token to be added to IlpCredentials
+   * @return a new instance of IlpCredentials, with an Authorization header
+   */
+  public static build(token?: string): IlpCredentials {
```

Comment:
Nope, not really sure.  Guess my head is still in Java-land where we like builders.  I'll change to a constructor

Reply:
On second thought, for grpc-web calls, `metadata` has to be of type `Metadata | undefined` because auth could come from a cookie.  I could do this check in the network client, but kind of like it in a static `build` method.  

We don't have the same problem for the node client, but I feel like we should keep them more or less the same